### PR TITLE
[DOC-3069] Mark as failure

### PR DESCRIPTION
**Pull request by rashminandasahoo**
New topic to explain the `mark as failure` option for pipeline stages.

## What Type of PR is This?

- [ ] Issue
- [ x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ x] Tested Locally
- [x] *Optional* Screen Shoot. 
mark-as-failed-option

PR Preview: https://6461f8d6475a96005410e8a9--harness-developer.netlify.app/

### CODE REVIEW

File: docs/platform/8_Pipelines/mark-as-failed.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+---
+title: Mark stage as failed
+description: Mark a stage as failed during execution.
+sidebar_position: 9
+---
+
+You can mark a specific stage as failed during pipeline execution by selecting the `Mark Stage As Failed` option.
```
Comment (SudarshanPurohit) [inline docs/platform/8_Pipelines/mark-as-failed.md:7]: The label is "Mark as Failure", I believe. @sahilhindwani-harness ?

File: docs/platform/8_Pipelines/mark-as-failed.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+This lets you do the following: 
+- Abort a step that is running.
+- Trigger rollback after the step is aborted.
```
Comment (SudarshanPurohit) [inline docs/platform/8_Pipelines/mark-as-failed.md:10]: Basically we can trigger whatever failure strategy is configured, including Rollback or Retry or anything else.

File: docs/platform/8_Pipelines/mark-as-failed.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+## Configure failure strategy
+
+To configure failure strategy for a stage marked as failure: 
+
+1. Select your pipeline and then select the stage for which you want to configure failure strategy.
+2. Select **Advanced**.
+3. In **Failure Strategy**, select **Add**.
+4. Select **User Marked Failure** in **On failure of type**.
+5. Select desired action in **Perform Action**.
+   The action you select here gets triggered after you mark a stage as failed.
```
Comment (SudarshanPurohit) [inline docs/platform/8_Pipelines/mark-as-failed.md:42]: "You can select Rollback, Retry, or other options to be executed at this point."

File: docs/platform/8_Pipelines/mark-as-failed.md (hunk 1 above)
Comment (SudarshanPurohit) [reply docs/platform/8_Pipelines/mark-as-failed.md:7]: Apologies - no, it's "Mark Stage as Failed". My miss.,

File: docs/platform/8_Pipelines/mark-as-failed.md (hunk 3 above)
Comment (rashminandasahoo) [reply docs/platform/8_Pipelines/mark-as-failed.md:42]: @SudarshanPurohit **Perform Action** has all the options like Rollback, Retry, etc. Do we need to mention the options again specifically?

Comment (SudarshanPurohit) [reply docs/platform/8_Pipelines/mark-as-failed.md:42]: Yes, I see the previous line. Let's keep it as it is.

**Review by SudarshanPurohit — APPROVED** [review]

File: docs/platform/8_Pipelines/mark-as-failed.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+- You must have `Execute` pipeline permission to be able to mark as stage as failed.
+- The option `Mark Stage As Failed` is at a stage level in the execution phase. During the execution of parallel stages (like a matrix), you must either mark each one as a failure or let the other steps complete before applying the failure strategy.
+- Harness aborts any delegate task currently running after you mark a stage as failed.
+
+## Mark a stage as failed
+
+To mark a stage as failed during pipeline execution: 
+
+1. Select your pipeline and then select **Run**.
```
Comment (StrawberryPad) [inline docs/platform/8_Pipelines/mark-as-failed.md:22]:
```suggestion
1. Select your pipeline, and then select **Run**.
```

File: docs/platform/8_Pipelines/mark-as-failed.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+2. Enter runtime input if any and select **Run Pipeline**.
```
Comment (StrawberryPad) [inline docs/platform/8_Pipelines/mark-as-failed.md:23]:
```suggestion
2. Enter runtime input, if any, and then select **Run Pipeline**.
```

File: docs/platform/8_Pipelines/mark-as-failed.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+3. Select **Mark Stage As Failed**.
+   
+   ![](./static/mark-as-failed-option.png)
+
+4. Select **Confirm**.
+   The **Error Summary** for the stage displays `User Initiated Failure`. The failure type of this stage is set to `USER_MARKED_FAILURE`.
```
Comment (StrawberryPad) [inline docs/platform/8_Pipelines/mark-as-failed.md:29]:
```suggestion
   
   The **Error Summary** for the stage displays `User Initiated Failure`. The failure type of this stage is set to `USER_MARKED_FAILURE`.
```

File: docs/platform/8_Pipelines/mark-as-failed.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+   ![](./static/error-summary.png)
+
+## Configure failure strategy
+
+To configure failure strategy for a stage marked as failure: 
+
+1. Select your pipeline and then select the stage for which you want to configure failure strategy.
```
Comment (StrawberryPad) [inline docs/platform/8_Pipelines/mark-as-failed.md:37]:
```suggestion
1. Select your pipeline, and then select the stage for which you want to configure failure strategy.
```

File: docs/platform/8_Pipelines/mark-as-failed.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+2. Select **Advanced**.
+3. In **Failure Strategy**, select **Add**.
+4. Select **User Marked Failure** in **On failure of type**.
+5. Select desired action in **Perform Action** like **Rollback**, **Retry**.
```
Comment (StrawberryPad) [inline docs/platform/8_Pipelines/mark-as-failed.md:41]:
```suggestion
5. Select desired action in **Perform Action**, for example, **Rollback** or **Retry**.
```

File: docs/platform/8_Pipelines/mark-as-failed.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+   The action you select here gets triggered after you mark a stage as failed.
+6. Select **Save**.
+
+   ![](./static/failure%20strategy.png)
+
+
+:::important
+If you mark a stage as failed whose failure strategy is not defined, Harness triggers the default failure strategy.
```
Comment (StrawberryPad) [inline docs/platform/8_Pipelines/mark-as-failed.md:49]:
```suggestion
If you mark a stage as failed, and its failure strategy is not defined, Harness triggers the default failure strategy.
```

**Review by StrawberryPad — APPROVED** [review]

**Pull request merged**